### ISSUE TITLE
ISO 19101-1:2014: Examples of “conceptual formalism”

### ISSUE THREAD

**Issue by heidivanparys** [ISO-TC211/StandardsTracker#459]
ISO 19101-1:2014 states in the term entry for [conceptual formalism](https://www.iso.org/obp/ui#iso:std:iso:19101:-1:ed-1:v1:en:term:4.1.4) that the UML metamodel and the EXPRESS metamodel are examples of “conceptual formalism”, however, these are examples of metamodels. A correct example of conceptual formalism would be “object-oriented modelling”, see also clause 7.2, stating that “The applicable conceptual formalism for the ISO geographic information standards shall be object-oriented modelling”.

**Comment by ronaldtse** [comment]
Regarding EXPRESS, there is the EXPRESS language and an EXPRESS metamodel (at OMG) developed to represent the EXPRESS language. 

As for what constitutes "conceptual formalism", according to its definition of "set of modelling concepts used to describe a conceptual model", it seems to be the primitives that are used to build a conceptual model, eg. Relationships/associations, properties/attributes, entities/classes etc. 

The metamodels of UML and EXPRESS seem to describe those. Do they not?

**Comment by heidivanparys** [comment]
I wasn't involved in the development of 19101-1, so I don't really know the background of this.

So “[conceptual formalism](https://www.iso.org/obp/ui#iso:std:iso:19101:-1:ed-1:v1:en:term:4.1.4)” is the exact same concept as “[metamodel](https://www.iso.org/obp/ui#iso:std:iso:19103:ed-1:v1:en:term:4.22)”, used in 19103 and 19109?

Meaning that these sentences from ISO 19109:

_[...] the GFM is a metamodel for definition of features that is also used to define the structure of feature catalogues. The chosen CSL [...] is the metamodel for an application schema._

could be replaced by 

_[...] the GFM is a conceptual formalism for definition of features that is also used to define the structure of feature catalogues. The chosen CSL [...] is the conceptual formalism for an application schema._

?

If so, which of those two terms should be used in the future (given that both 19103 and 19109 are currently under revision)?

**Comment by ronaldtse** [comment]
I think you're right, these two concepts are highly associated, except that:
* "metamodel" describes "other models" but "conceptual formalism" only describes "conceptual models"
* so "metamodel" may be more generic

In fact the term "conceptual formalism" seems more like to be a description of a method (an "-ism") not conveying a "set of concepts". If there was a choice I'd prefer keeping "metamodel" instead of the rather rare phrase "conceptual formalism"...

**Comment by heidivanparys** [comment]
Yes, I agree, “conceptual formalism” also sounds more like a method, an approach, a paradigm to me. I guess other examples, at least when it comes to data, would be relational modelling and graph modelling.

**Comment by JanHjelmager** [comment]
Based on the discussion above. I think we can close the issue, as an agreement/consensus  have been reached.

**Comment by heidivanparys** [comment]
@JanHjelmager @ReesePlews This would still have to be implemented in ISO 19101-1:2014, alternatively in the new terminology standard, so I'm concerned that closing this now will make us forget about it. How will this be dealt with?

**Comment by ReesePlews** [comment]
i agree with @heidivanparys if the ticket is closed i will most definitely forget about these things the next time the discussion roles around. what about making a long term ticket for items that were discussed but need to "stick-around" until they are addressed during revision project work, etc.  @JanHjelmager would that be possible?

**Comment by PeterParslow** [comment]
If the current consensus is on what to do, but it hasn't been done, then I think the issue should remain open!

**Comment by JanHjelmager** [comment]
I reopen the issue :-)

**Comment by ronaldtse** [comment]
I have a suggestion -- why not create a Project or a Milestone for a ISO 19101-1 revision or the new terminology standard, and move this issue into it? Then when the project or milestone becomes active, the project leads can just take over all those tasks.

**Comment by heidivanparys** [comment]
> I have a suggestion -- why not create a Project or a Milestone for a ISO 19101-1 revision or the new terminology standard, and move this issue into it? Then when the project or milestone becomes active, the project leads can just take over all those tasks.

Good idea. Of the two, perhaps using [milestones](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is the best solution for this standards tracker? As a revision project probably will create its own project to manage issues, and will have copies of the issues originating from the standards tracker in there.

**Comment by ronaldtse** [comment]
Thanks @heidivanparys . I just realised that an Project may have additional benefits -- a GH Project can be org-wide, and when the standardization project gets activated, they can take over those tasks by transferring them into the project repo. A milestone is locked inside a repo it was created in (a milestone represents versions/points of the current repo, StandardsTracker).

**Comment by ronaldtse** [comment]
Oh, we can certain just copy the issues in. Indeed maybe we should keep these issues as a public record, but we sure want to close them when they get completed in private!
